devicetypes/smartthings/fibaro-rgbw-controller.src/fibaro-rgbw-controller.groovy

```diff
@@ -26,562 +26,233 @@
-    log.debug "firmwareId:     ${cmd.firmwareId}"
-    log.debug "manufacturerId: ${cmd.manufacturerId}"
-}
-
-def zwaveEvent(physicalgraph.zwave.commands.colorcontrolv1.CapabilityReport cmd, Map item1) {
-
-	log.debug "In CapabilityReport"
-}
-
-def createEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelCmdEncap cmd, Map item1) {
-	def encapsulatedCommand = cmd.encapsulatedCommand([0x26: 1, 0x30: 2, 0x32: 2, 0x33: 2]) // can specify command class versions here like in zwave.parse
+def zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelCmdEncap cmd) {
+	def encapsulatedCommand = cmd.encapsulatedCommand() // can specify command class versions here like in zwave.parse
 	//log.debug ("Command from endpoint ${cmd.sourceEndPoint}: ${encapsulatedCommand}")
 	if ((cmd.sourceEndPoint >= 1) && (cmd.sourceEndPoint <= 5)) { // we don't need color report
-    	//don't do anything
-    } else {
-    	if (encapsulatedCommand) {
+		// ignored (these are rgb reports in a non-standard format)
+	} else {
+		if (encapsulatedCommand) {
 			zwaveEvent(encapsulatedCommand)
-        }
+		}
 	}
 }
 
-def createEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd, Map item1) {
-     def result = doCreateEvent(cmd, item1)
-     for (int i = 0; i < result.size(); i++) {
-          result[i].type = "physical"
-     }
-     result
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd) {
+	dimmerEvents(cmd)
 }
 
-def createEvent(physicalgraph.zwave.commands.basicv1.BasicSet cmd, Map item1) {
-     def result = doCreateEvent(cmd, item1)
-     for (int i = 0; i < result.size(); i++) {
-          result[i].type = "physical"
-     }
-     result
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicSet cmd) {
+	dimmerEvents(cmd)
 }
 
-def createEvent(physicalgraph.zwave.commands.sensormultilevelv2.SensorMultilevelReport cmd, Map item1) {
-    def result = [:]
-    if ( cmd.sensorType == 4 ) { //power level comming in
-   		result.name = "power"
-    	result.value = cmd.scaledSensorValue
-    	result.descriptionText = "$device.displayName power usage is ${result.value} watt(s)"
-        result.isStateChange
-        sendEvent(name: result.name, value: result.value, displayed: false)
-    }
-    result
+def zwaveEvent(physicalgraph.zwave.commands.switchmultilevelv3.SwitchMultilevelReport cmd) {
+	dimmerEvents(cmd)
 }
 
-def createEvent(physicalgraph.zwave.commands.switchmultilevelv1.SwitchMultilevelStartLevelChange cmd, Map item1) {
-     []
+def zwaveEvent(physicalgraph.zwave.commands.switchmultilevelv3.SwitchMultilevelSet cmd) {
+	dimmerEvents(cmd)
 }
 
-def createEvent(physicalgraph.zwave.commands.switchmultilevelv1.SwitchMultilevelStopLevelChange cmd, Map item1) {
-     [response(zwave.basicV1.basicGet())]
+private dimmerEvents(physicalgraph.zwave.Command cmd) {
+	def value = (cmd.value ? "on" : "off")
+	def result = [createEvent(name: "switch", value: value)]
+	if (cmd.value && cmd.value <= 100) {
```

Comment:
how about value 0? Are we not generate value for 0 so it be switched back to that value when enabled again?

Reply:
This was cribbed directly from another zwave dimmer dth. Typically we set the level to 0 with zwave devices and leave it to the previous value with zigbee. Why it is like this is anyone's guess, but I'm not sure it's time to buck the trend yet.